src/index.js

```diff
@@ -4,19 +4,19 @@ export const BEGIN = '@@optimist/BEGIN';
 export const COMMIT = '@@optimist/COMMIT';
 export const REVERT = '@@optimist/REVERT';
 
-export const ensureState = state => {
-  if (Map.isMap(state)) {
-    if (List.isList(state.get('history'))) {
-      return state.get('current');
-    }
-  }
-  return state;
-};
+// this marker is used to identify our wrapped state. The marker is designed to prevent false positives
+// that might occur if we checked for the existence of other properties. It is exported for testing.
+export const STATE_MARKER = '__redux-optimistic-ui';
```

Comment:
Do you think this is an OK approach? 

I originally had the `isOptimistState` check as `const isOptimistState = state => Map.isMap(state) && List.isList(state.get('history')) && state.has('current');` however my thinking was there was always some very rare chance that someone's reducers might actually look like that (especially when we move to native JS objects). Having this unique marker means we only need to check for that marker instead.